conformance/lang/expressions/multiplicative-expr/division.balt

```diff
@@ -4033,3 +4033,851 @@ Description: Test integer division panic at module-level.
+    io:println(25.0E17f / 0.9E-10); // @output 2.7777777777777776E28
+
+    a = -25.0E17f;
+    b = 0.9E-10f;
+    c = a / b;
+    io:println(c); // @output -2.7777777777777776E28
+    io:println(-25.0E17f / 0.9E-10); // @output -2.7777777777777776E28
+
+    a = 17.0E12f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output Infinity
+    io:println(17.0E12f / 0.0f); // @output Infinity
+
+    a = -17.0E12f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output -Infinity
+    io:println(-17.0E12f / 0.0f); // @output -Infinity
+
+    a = 11.0E12f;
+    b = -0.0f;
+    c = a / b;
+    io:println(c); // @output -Infinity
+    io:println(11.0E12f / -0.0f); // @output -Infinity
+
+    a = float:Infinity;
+    b = float:Infinity;
+    c = a / b;
+    io:println(c); // @output NaN
+    io:println(float:Infinity / float:Infinity); // @output NaN
+
+    a = 0.0f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output NaN
+    io:println(0.0f / 0.0f); // @output NaN
+
+    a = -0.0f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output NaN
+    io:println(-0.0f / 0.0f); // @output NaN
+
+    a = float:PI;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output Infinity
+    io:println(float:PI / 0.0f); // @output Infinity
+
+    a = float:PI;
+    b = 0.0f;
+    c = b / a;
+    io:println(c); // @output 0.0
+    io:println(0.0f / float:PI); // @output 0.0
+}
+
+Test-Case: output
+Description: Test division scenarios for different nillable float values.
+Labels: multiplicative-expr, DecimalFloatingPointNumber, float, FloatTypeSuffix, optional-type, numeric-nil-lifting, unary-minus, nil-literal, value:toBalString
+
+function init() {
+    float? a = 2.5E-17;
+    float? b = 2.5E+17;
+    float? c = b / a;
+    io:println(c); // @output 1.0000000000000001E34
+
+    a = 1.423223E6;
+    b = 2.34413E2;
+    c = a / b;
+    io:println(c); // @output 6071.433751541083
+
+    a = 25.0E17f;
+    b = 0.9E-10f;
+    c = a / b;
+    io:println(c); // @output 2.7777777777777776E28
+
+    a = -25.0E17f;
+    b = 0.9E-10f;
+    c = a / b;
+    io:println(c); // @output -2.7777777777777776E28
+
+    a = 17.0E12f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output Infinity
+
+    a = -17.0E12f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output -Infinity
+
+    a = 11.0E12f;
+    b = -0.0f;
+    c = a / b;
+    io:println(c); // @output -Infinity
+
+    a = 0.0f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output NaN
+
+    a = -0.0f;
+    b = 0.0f;
+    c = a / b;
+    io:println(c); // @output NaN
+
+    a = 11.0E12f;
+    b = ();
+    c = a / b;
+    io:println(c.toBalString()); // @output ()
+
+    a = ();
+    b = 13.0E121f;
+    c = a / b;
+    io:println(c.toBalString()); // @output ()
+}
+
+Test-Case: output
+Description: Test division by infinity for different float values.
+Labels: multiplicative-expr, DecimalFloatingPointNumber, float, FloatTypeSuffix, unary-minus, float:Infinity, float:NaN
```

Comment:
Need to use `infinity` label for `float:Infinity`. Please check other usages too.